crates/oxc_syntax/src/identifier.rs

```diff
@@ -39,6 +42,26 @@ pub fn is_irregular_whitespace(c: char) -> bool {
     )
 }
 
+const NEL: char = '\u{85}';
+const OGHAM_SPACE_MARK: char = '\u{1680}';
+const EN_QUAD: char = '\u{2000}';
+const ZWSP: char = '\u{200b}';
+
+// Narrow No-Break Space
+const NNBSP: char = '\u{202f}';
+
+// Medium Mathematical Space
+const MMSP: char = '\u{205f}';
+const IDEOGRAPHIC_SPACE: char = '\u{3000}';
+
+// https://github.com/microsoft/TypeScript/blob/b8e4ed8aeb0b228f544c5736908c31f136a9f7e3/src/compiler/scanner.ts#L556
+pub fn is_white_space_single_line(c: char) -> bool {
+    // Note: nextLine is in the Zs space, and should be considered to be a whitespace.
+    // It is explicitly not a line-break as it isn't in the exact set specified by EcmaScript.
+    matches!(c, |SP| TAB | VT | FF | NBSP | NEL | OGHAM_SPACE_MARK | EN_QUAD
+        ..=ZWSP | NNBSP | MMSP | IDEOGRAPHIC_SPACE | ZWNBSP)
```

Comment:
Style nit (remove the `|` before `SP`, and then rustfmt does the rest):

```suggestion
    matches!(
        c,
        SP | TAB | VT | FF | NBSP | NEL | OGHAM_SPACE_MARK | EN_QUAD
            ..=ZWSP | NNBSP | MMSP | IDEOGRAPHIC_SPACE | ZWNBSP
    )
```